src/components/flat-table/flat-table.component.tsx

```diff
@@ -181,105 +215,57 @@ const FlatTable = ({
-          flexDirection="column"
-          justifyContent={
-            hasStickyFooter || height ? "space-between" : undefined
-          }
-          tabIndex="0"
-          role="region"
-          overflowX={width ? "hidden" : undefined}
-          width={width}
-          hasStickyFooter={hasStickyFooter}
-          hasVerticalScrollbar={hasVerticalScrollbar}
-          hasHorizontalScrollbar={hasHorizontalScrollbar}
-          hasFooter={!!footer}
-          firstColRowSpanIndex={firstColRowSpanIndex}
-          lastColRowSpanIndex={lastColRowSpanIndex}
-          onKeyDown={handleKeyDown}
-          {...rest}
+    <StyledFlatTableWrapper
+      ref={wrapperRef}
+      data-component="flat-table-wrapper"
+      isInSidebar={isInSidebar}
+      hasStickyHead={hasStickyHead}
+      colorTheme={colorTheme}
+      minHeight={minHeight}
+      overflowY={
+        !isInSidebar && (hasStickyHead || hasStickyFooter) ? "auto" : undefined
+      }
+      height={addDefaultHeight && !hasMaxHeight ? "99%" : height}
+      maxHeight={hasMaxHeight ? "100%" : undefined}
+      display="flex"
+      flexDirection="column"
+      justifyContent={hasStickyFooter || height ? "space-between" : undefined}
+      tabIndex={0}
+      role="region"
+      overflowX={width ? "hidden" : undefined}
+      width={width}
+      hasStickyFooter={hasStickyFooter}
+      hasVerticalScrollbar={hasVerticalScrollbar}
+      hasHorizontalScrollbar={hasHorizontalScrollbar}
+      footer={!!footer}
+      firstColRowSpanIndex={firstColRowSpanIndex}
+      lastColRowSpanIndex={lastColRowSpanIndex}
+      onKeyDown={handleKeyDown}
+      {...rest}
+    >
+      <StyledTableContainer overflowX={overflowX} width={width}>
+        <StyledFlatTable
+          ref={tableRef}
+          data-component="flat-table"
+          {...tableStylingProps}
         >
-          <StyledTableContainer overflowX={overflowX} width={width}>
-            <StyledFlatTable
-              ref={tableRef}
-              data-component="flat-table"
-              {...tableStylingProps}
-            >
-              {caption ? <caption>{caption}</caption> : null}
-              <FlatTableThemeContext.Provider
-                value={{ colorTheme, size, setSelectedId, selectedId }}
-              >
-                {children}
-              </FlatTableThemeContext.Provider>
-            </StyledFlatTable>
-            {footer && (
-              <StyledFlatTableFooter hasStickyFooter={hasStickyFooter}>
-                {footer}
-              </StyledFlatTableFooter>
-            )}
-          </StyledTableContainer>
-        </StyledFlatTableWrapper>
-      )}
-    </DrawerSidebarContext.Consumer>
+          {caption ? <caption>{caption}</caption> : null}
+          <FlatTableThemeContext.Provider
+            value={{ colorTheme, size, setSelectedId, selectedId }}
+          >
+            {children}
+          </FlatTableThemeContext.Provider>
+        </StyledFlatTable>
+        {footer && (
```

Comment:
note: remember you'll need to move this outside of the `StyledFlatTableContainer` to avoid introducing the bug with stick footer.